packages/odyssey-react-mui/src/Typography.tsx

```diff
@@ -0,0 +1,165 @@
+/*!
+ * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {
+  Typography as MuiTypography,
+  TypographyProps as MuiTypographyProps,
+} from "@mui/material";
+import { ElementType, ReactNode } from "react";
+
+export const typographyVariantValues = {
+  h1: "h1",
+  h2: "h2",
+  h3: "h3",
+  h4: "h4",
+  h5: "h5",
+  h6: "h6",
+  body: "body1",
+  caption: "subtitle1",
+  legend: "legend",
+};
+
+export type TypographyProps = {
+  children: ReactNode;
+  classes?: object;
+  component?: ElementType;
+  variant?: keyof typeof typographyVariantValues;
+};
+
+export const Typography = ({
+  children,
+  classes,
+  component,
+  variant = "body",
+}: TypographyProps) => {
+  if (!component) {
+    component = variant;
+    if (variant === "body") {
+      component = "p";
+    }
+    if (variant === "caption") {
+      component = "h6";
+    }
+  }
+
+  return (
+    <MuiTypography
+      children={children}
+      classes={classes}
+      component={component}
+      variant={
+        typographyVariantValues[variant] as MuiTypographyProps["variant"]
```

Comment:
Needed to use `as` to appease the type checker, but I'm okay with it here because we have control/visibility over both `typographyVariantValues` and `MuiTypographyProps`, so we know they'll match.